java/fury-core/src/main/java/org/apache/fury/serializer/JdkProxySerializer.java

```diff
@@ -23,13 +23,27 @@
 import java.lang.reflect.Proxy;
 import org.apache.fury.Fury;
 import org.apache.fury.memory.MemoryBuffer;
+import org.apache.fury.resolver.RefResolver;
+import org.apache.fury.util.Platform;
 import org.apache.fury.util.Preconditions;
 import org.apache.fury.util.ReflectionUtils;
 
 /** Serializer for jdk {@link Proxy}. */
 @SuppressWarnings({"rawtypes", "unchecked"})
 public class JdkProxySerializer extends Serializer {
 
+  // Make offset compatible with graalvm native image.
+  private static final long PROXY_HANDLER_FIELD_OFFSET;
+
+  static {
+    try {
+      // Make offset compatible with graalvm native image.
+      PROXY_HANDLER_FIELD_OFFSET = Platform.objectFieldOffset(Proxy.class.getDeclaredField("h"));
+    } catch (NoSuchFieldException e) {
+      throw new RuntimeException(e);
```

Comment:
The name `h` is not a common abbr, could we use offset from a field with type `InvocationHandler` in catch block? We can add a method in `ReflectionUtils` which check a class has only one filed with specified type an return the offset of that type

Reply:
I tried to implement it this way in my latest commit ;)